docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,471 @@
+
+Please see more details in [Python User Guide](https://bigdl.readthedocs.io/en/latest/doc/UserGuide/python.html).
+
+## 2.3 Notes for CDH Users
+* For CDH user, the value of environment variable `HADOOP_CONF_DIR` should be `/etc/hadoop/conf` by default.
+
+* The __Client Node__ (where you submit applications) may have already installed a different version of Spark then the one installed with BigDL. To avoid conflicts, unset all Spark-related environment variables (you may use use `env | grep SPARK` to find all of them):
+    ```bash
+    unset SPARK_HOME
+    unset SPARK_VERSION
+    unset ...
+    ```
+
+* You also need to properly export the location of `libhdfs.so` if needed. For instance, when using pyarrow, set up the environment variable `ARROW_LIBHDFS_DIR` on the __Client Node__ (where you submit the program to the YARN cluster) as follows:
+    ```bash
+    export ARROW_LIBHDFS_DIR=/opt/cloudera/parcels/CDH-${CHD_VERSION}/lib64/
+    ```
+
+# 3. Prepare Dataset 
+To run the example on YARN, you should upload the Fashion-MNIST dataset to a distributed storage (such as HDFS or S3).   
+
+First, please download the Fashion-MNIST dataset manually on your __Client Node__ (where you submit the program to YARN).
+```bash
+# PyTorch official dataset download link
+git clone https://github.com/zalandoresearch/fashion-mnist.git
+
+mv /path/to/fashion-mnist/data/fashion /path/to/local/data/FashionMNIST/raw 
+```
+Then upload it to a distributed storage.
+```bash
+# Upload to HDFS
+hdfs dfs -put /path/to/local/data/FashionMNIST hdfs://url:port/path/to/remote/data
+```
+
+# 4. Prepare Custom Modules
+Spark allows to upload Python files(`.py`), and zipped Python packages(`.zip`) to the executors by setting `--py-files` option in Spark scripts or `extra_python_lib` in `init_orca_context`. 
+
+The example depends on few Python files, you could upload them to YARN on __Client Node__ (where you submit the program to YARN) directly.
+* When using `python` command, please specify `extra_python_lib` in `init_orca_context`.
+    ```python
+    init_orca_context(cluster_mode, cores, memory, num_nodes,
+                      extra_python_lib="/path/to/model.py")
+    ```
+
+    Please see more details in [OrcaContext Python Dependencies](https://bigdl.readthedocs.io/en/latest/doc/Orca/Overview/orca-context.html#python-dependencies).
+
+* When using `bigdl-submit` or `spark-submit` script, please specify `--py-files` option.
+    ```bash
+    --py-files /path/to/model.py
+    ```
+
+    Please see more details in [Spark Submitting Applications](https://spark.apache.org/docs/latest/submitting-applications.html). 
+
+## Optional (Use Zipped Packages)
+If your program depends on a nested directory of Python files, you could package them into a zipped packages on the __Client Node__ (where you submit applications) instead. 
+
+For example, `example` is the nested directory including `model.py` needed for this program.
```

Comment:
use FashionMNIST as an example; does the user need to do anything special in his or her code?

Reply:
> use FashionMNIST as an example; does the user need to do anything special in his or her code?

I mentioned it in the "Note", users need to import modules like `from example.model import model_creator`.